### ISSUE TITLE
Specific page parts for templates

### ISSUE THREAD

**Issue by sintro** [refinery/refinerycms#3152]
As I can understand, there are still no fine solution to use this kind of feature from initializer of page.rb
```
  # config.types.register :home do |home|
  #   home.parts = %w[intro body]
  # end
```
So I tried to implement this https://gist.github.com/jipiboily/2338880 for my project. 
It is sad that this feature is not implemented in Core (or something even more handy) still. But the worse thing is that even solution from https://gist.github.com/jipiboily/2338880 does not work. The problem is:
```
TypeError in Refinery::Admin::PagesController#new
no implicit conversion of Symbol into Integer
Extracted source (around line 15):
             
        @page = Page.new(new_page_params)
        Pages.default_parts_for(@page).each_with_index do |page_part, index|
          @page.parts << PagePart.new(:title => page_part[:title], :slug => page_part[:slug], :position => index)
        end
      end

```
It is trying to acess ``page_part[:key]``, when it is declared as simple strings in config.types.register. It is impossible to declare this parts as hashes (like it is done now for default_parts), because now the method ``Page.default_parts_for`` will fail thanks to .parts.map(&:titleize).

So the possible solutions for the problem are:
1. At least implement whole this feature in Core with some handy way.
2. Fix ``Page.default_parts_for`` so it will be able to swallow custom parts defined as hashed (like it really should be)

**Comment by sintro** [comment]
For now I fixed the problem by creating the pages/pages_decorator.rb with content
```
Refinery::Pages.module_eval do
  def self.default_parts_for(page)
    return default_parts unless page.view_template.present?

    types.find_by_name(page.view_template).parts
  end
end
```
And by defining the custom page parts as array of hashes:
```
  config.types.register :contacts do |page|
    page.parts = [{title: 'Body', slug: 'body'}, {title: 'Address', slug: 'address'}]
  end
```

**Comment by bricesanchez** [comment]
That's a nice feature, could create a pull request ?

**Comment by bricesanchez** [comment]
🆙 :)

**Comment by sintro** [comment]
Sure, I just have no time right now to make everything looks shiny and ready to PR, I also don`t think that will be able to make nice tests, may be someone else will finish them. 
I also think about separating this to side "refinery-custom_page_types" gem (may be it is worth to do)

**Comment by bricesanchez** [comment]
Closed in favor of #3221
